<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Versatile BedGraph reader. — read_bedgraphs • methrix</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Versatile BedGraph reader. — read_bedgraphs" />
<meta property="og:description" content="Versatile BedGraph reader." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">methrix</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/methrix.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/02-Import.html">01: Reading bedgraph files</a>
    </li>
    <li>
      <a href="../articles/03-QC_reports.html">02: QC and html reports</a>
    </li>
    <li>
      <a href="../articles/04-DMRs.html">03: Indentifying DMRs</a>
    </li>
    <li>
      <a href="../articles/05-Large_cohorts.html">05: Working with large cohorts</a>
    </li>
    <li>
      <a href="../articles/06-Export.html">06: Exporting to standard file formats</a>
    </li>
    <li>
      <a href="../articles/07-Reader_FAQ.html">FAQ: Detailed documentation of the read_bedgraph function.</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Versatile BedGraph reader.</h1>
    
    <div class="hidden name"><code>read_bedgraphs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Versatile BedGraph reader.</p>
    </div>

    <pre class="usage"><span class='fu'>read_bedgraphs</span><span class='op'>(</span>
  files <span class='op'>=</span> <span class='cn'>NULL</span>,
  pipeline <span class='op'>=</span> <span class='cn'>NULL</span>,
  zero_based <span class='op'>=</span> <span class='cn'>TRUE</span>,
  stranded <span class='op'>=</span> <span class='cn'>FALSE</span>,
  collapse_strands <span class='op'>=</span> <span class='cn'>FALSE</span>,
  ref_cpgs <span class='op'>=</span> <span class='cn'>NULL</span>,
  ref_build <span class='op'>=</span> <span class='cn'>NULL</span>,
  contigs <span class='op'>=</span> <span class='cn'>NULL</span>,
  vect <span class='op'>=</span> <span class='cn'>FALSE</span>,
  vect_batch_size <span class='op'>=</span> <span class='cn'>NULL</span>,
  coldata <span class='op'>=</span> <span class='cn'>NULL</span>,
  chr_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  start_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  end_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  beta_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  M_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  U_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  strand_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  cov_idx <span class='op'>=</span> <span class='cn'>NULL</span>,
  synced_coordinates <span class='op'>=</span> <span class='cn'>FALSE</span>,
  n_threads <span class='op'>=</span> <span class='fl'>1</span>,
  h5 <span class='op'>=</span> <span class='cn'>FALSE</span>,
  h5_dir <span class='op'>=</span> <span class='cn'>NULL</span>,
  h5temp <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>files</th>
      <td><p>bedgraph files.</p></td>
    </tr>
    <tr>
      <th>pipeline</th>
      <td><p>Default NULL. Currently supports "Bismark_cov", "MethylDackel", "MethylcTools", "BisSNP", "BSseeker2_CGmap"
If not known use idx arguments for manual column assignments.</p></td>
    </tr>
    <tr>
      <th>zero_based</th>
      <td><p>Are bedgraph regions zero based ? Default TRUE</p></td>
    </tr>
    <tr>
      <th>stranded</th>
      <td><p>Default FALSE</p></td>
    </tr>
    <tr>
      <th>collapse_strands</th>
      <td><p>If TRUE collapses CpGs on different crick strand into watson. Deafult FALSE</p></td>
    </tr>
    <tr>
      <th>ref_cpgs</th>
      <td><p>BSgenome object, or name of the installed BSgenome package, or an output from <code><a href='extract_CPGs.html'>extract_CPGs</a></code>.
Example: BSgenome.Hsapiens.UCSC.hg19</p></td>
    </tr>
    <tr>
      <th>ref_build</th>
      <td><p>reference genome for bedgraphs. Default NULL. Only used for additional details. Doesnt affect in any way.</p></td>
    </tr>
    <tr>
      <th>contigs</th>
      <td><p>contigs to restrict genomic CpGs to. Default all autosomes and allosomes - ignoring extra contigs.</p></td>
    </tr>
    <tr>
      <th>vect</th>
      <td><p>To use vectorized code. Default FALSE. Set to TRUE if you don't have large number of BedGraph files.</p></td>
    </tr>
    <tr>
      <th>vect_batch_size</th>
      <td><p>Default NULL. Process samples in batches. Applicable only when vect = TRUE</p></td>
    </tr>
    <tr>
      <th>coldata</th>
      <td><p>An optional DataFrame describing the samples. Row names, if present, become the column names of the matrix.
If NULL, then a DataFrame will be created with basename of files used as the row names.</p></td>
    </tr>
    <tr>
      <th>chr_idx</th>
      <td><p>column index for chromosome in bedgraph files</p></td>
    </tr>
    <tr>
      <th>start_idx</th>
      <td><p>column index for start position in bedgraph files</p></td>
    </tr>
    <tr>
      <th>end_idx</th>
      <td><p>column index for end position in bedgraph files</p></td>
    </tr>
    <tr>
      <th>beta_idx</th>
      <td><p>column index for beta values in bedgraph files</p></td>
    </tr>
    <tr>
      <th>M_idx</th>
      <td><p>column index for read counts supporting Methylation in bedgraph files</p></td>
    </tr>
    <tr>
      <th>U_idx</th>
      <td><p>column index for read counts supporting Un-methylation in bedgraph files</p></td>
    </tr>
    <tr>
      <th>strand_idx</th>
      <td><p>column index for strand information in bedgraph files</p></td>
    </tr>
    <tr>
      <th>cov_idx</th>
      <td><p>column index for total-coverage in bedgraph files</p></td>
    </tr>
    <tr>
      <th>synced_coordinates</th>
      <td><p>Are the start and end coordinates of a stranded
bedgraph are synchronized between + and - strands? Possible values: FALSE (default),
TRUE if the start coordinates are the start coordinates of the C on the plus strand.</p></td>
    </tr>
    <tr>
      <th>n_threads</th>
      <td><p>number of threads to use. Default 1.
Be-careful - there is a linear increase in memory usage with number of threads. This option is does not work with Windows OS.</p></td>
    </tr>
    <tr>
      <th>h5</th>
      <td><p>Should the coverage and methylation matrices be stored as 'HDF5Array'</p></td>
    </tr>
    <tr>
      <th>h5_dir</th>
      <td><p>directory to store H5 based object</p></td>
    </tr>
    <tr>
      <th>h5temp</th>
      <td><p>temporary directory to store hdf5</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Be little chatty ? Default TRUE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code><a href='methrix-class.html'>methrix</a></code></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Reads BedGraph files and generates methylation and coverage matrices.
Optionally arrays can be serialized as on-disk HDFS5 arrays.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>bdg_files</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.files.html'>list.files</a></span><span class='op'>(</span>path <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>'extdata'</span>, package <span class='op'>=</span> <span class='st'>'methrix'</span><span class='op'>)</span>,
pattern <span class='op'>=</span> <span class='st'>'*\\.bedGraph\\.gz$'</span>, full.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>hg19_cpgs</span> <span class='op'>=</span> <span class='fu'>methrix</span><span class='fu'>::</span><span class='fu'><a href='extract_CPGs.html'>extract_CPGs</a></span><span class='op'>(</span>ref_genome <span class='op'>=</span> <span class='st'>'BSgenome.Hsapiens.UCSC.hg19'</span><span class='op'>)</span>
<span class='va'>meth</span> <span class='op'>=</span> <span class='fu'>methrix</span><span class='fu'>::</span><span class='fu'>read_bedgraphs</span><span class='op'>(</span> files <span class='op'>=</span> <span class='va'>bdg_files</span>, ref_cpgs <span class='op'>=</span> <span class='va'>hg19_cpgs</span>,
chr_idx <span class='op'>=</span> <span class='fl'>1</span>, start_idx <span class='op'>=</span> <span class='fl'>2</span>, M_idx <span class='op'>=</span> <span class='fl'>3</span>, U_idx <span class='op'>=</span> <span class='fl'>4</span>,
stranded <span class='op'>=</span> <span class='cn'>FALSE</span>, zero_based <span class='op'>=</span> <span class='cn'>FALSE</span>, collapse_strands <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='op'>}</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Anand Mayakonda, Reka Toth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


