<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Removes CpG sites from the object if they overlap with common SNPs — remove_snps • methrix</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Removes CpG sites from the object if they overlap with common SNPs — remove_snps" />
<meta property="og:description" content="Removes CpG sites from the object if they overlap with common SNPs" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">methrix</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/methrix.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/02-Import.html">01: Reading bedgraph files</a>
    </li>
    <li>
      <a href="../articles/03-QC_reports.html">02: QC and html reports</a>
    </li>
    <li>
      <a href="../articles/04-DMRs.html">03: Indentifying DMRs</a>
    </li>
    <li>
      <a href="../articles/05-Large_cohorts.html">05: Working with large cohorts</a>
    </li>
    <li>
      <a href="../articles/06-Export.html">06: Exporting to standard file formats</a>
    </li>
    <li>
      <a href="../articles/07-Reader_FAQ.html">FAQ: Detailed documentation of the read_bedgraph function.</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Removes CpG sites from the object if they overlap with common SNPs</h1>
    
    <div class="hidden name"><code>remove_snps.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Removes CpG sites from the object if they overlap with common SNPs</p>
    </div>

    <pre class="usage"><span class='fu'>remove_snps</span><span class='op'>(</span>
  <span class='va'>m</span>,
  populations <span class='op'>=</span> <span class='cn'>NULL</span>,
  maf_threshold <span class='op'>=</span> <span class='fl'>0.01</span>,
  reduce_filtering <span class='op'>=</span> <span class='cn'>FALSE</span>,
  forced <span class='op'>=</span> <span class='cn'>FALSE</span>,
  keep <span class='op'>=</span> <span class='cn'>FALSE</span>,
  n_chunks <span class='op'>=</span> <span class='fl'>1</span>,
  n_cores <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>m</th>
      <td><p><code><a href='methrix-class.html'>methrix</a></code> object</p></td>
    </tr>
    <tr>
      <th>populations</th>
      <td><p>Populations to use. Default is all.</p></td>
    </tr>
    <tr>
      <th>maf_threshold</th>
      <td><p>The frequency threshold, above which the SNPs will be removed. Default is 0.01</p></td>
    </tr>
    <tr>
      <th>reduce_filtering</th>
      <td><p>If TRUE, the SNPs with a MAF &lt; 0.1 will be evaluated and only the highly variable ones will be removed.
Default FALSE.</p></td>
    </tr>
    <tr>
      <th>forced</th>
      <td><p>the reduce_filtering is not recommended with less than 10 samples, but can be forced. Default is FALSE.</p></td>
    </tr>
    <tr>
      <th>keep</th>
      <td><p>Do you want to keep the sites that were filtered out? In this case, the function will return with a list of wo methrix objects.</p></td>
    </tr>
    <tr>
      <th>n_chunks</th>
      <td><p>Number of chunks to split the <code><a href='methrix-class.html'>methrix</a></code> object in case it is very large. Can only be used if input data is in HDF5 format. Default = 1.</p></td>
    </tr>
    <tr>
      <th>n_cores</th>
      <td><p>Number of parallel instances. Can only be used if input data is in HDF5 format. <code>n_cores</code> should be less than or equal to <code>n_chunks</code>. If <code>n_chunks</code> is not specified, then <code>n_chunks</code> is initialized to be equal to <code>n_cores</code>. Default = 1.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>methrix object or a list of methrix objects</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Takes <code><a href='methrix-class.html'>methrix</a></code> object and removes common SNPs. SNPs overlapping with a CpG site and have a minor allele frequency (MAF)
above a threshold in any of the populations used will be selected and the corresponding CpG sites will be removed from the <code><a href='methrix-class.html'>methrix</a></code> object.
With the reduce_filtering option, SNPs with MAP &lt; 0.1 will be further evaluated. If they show low variance in the dataset, there is probably no genotype
variability in the population, therefore the corresponding CpG site won't be removed. Please keep in mind that variance thresholds are</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'methrix_data'</span><span class='op'>)</span>
<span class='fu'>remove_snps</span><span class='op'>(</span>m <span class='op'>=</span> <span class='va'>methrix_data</span>, maf_threshold<span class='op'>=</span><span class='fl'>0.01</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in remove_snps(m = methrix_data, maf_threshold = 0.01): Packages MafDb.1Kgenomes.phase3.hs37d5 and/or GenomicScores not found.</span>
#&gt; <span class='error'>                 Please install them before proceed.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Anand Mayakonda, Reka Toth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


