<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>01: Reading bedgraph files • methrix</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="01: Reading bedgraph files">
<meta property="og:description" content="methrix">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">methrix</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/methrix.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/02-Import.html">01: Reading bedgraph files</a>
    </li>
    <li>
      <a href="../articles/03-QC_reports.html">02: QC and html reports</a>
    </li>
    <li>
      <a href="../articles/04-DMRs.html">03: Indentifying DMRs</a>
    </li>
    <li>
      <a href="../articles/05-Large_cohorts.html">05: Working with large cohorts</a>
    </li>
    <li>
      <a href="../articles/06-Export.html">06: Exporting to standard file formats</a>
    </li>
    <li>
      <a href="../articles/07-Reader_FAQ.html">FAQ: Detailed documentation of the read_bedgraph function.</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>01: Reading bedgraph files</h1>
            
      
      
      <div class="hidden name"><code>02-Import.Rmd</code></div>

    </div>

    
    
<div id="load-required-libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-libraries" class="anchor"></a>Load required libraries</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">methrix</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/seandavi/GEOquery">GEOquery</a></span><span class="op">)</span> <span class="co">#Only used for downloading example dataset </span></pre></div>
<div id="download-bedgraph-files" class="section level3">
<h3 class="hasAnchor">
<a href="#download-bedgraph-files" class="anchor"></a>Download bedgraph files</h3>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="va">data_dir</span> <span class="op">=</span> <span class="st">"~/Documents/GSE90553/"</span>
<span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">data_dir</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">filePaths</span> <span class="op">&lt;-</span>  <span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEOSuppFiles.html">getGEOSuppFiles</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GSM3039352"</span>, <span class="st">"GSM3039353"</span>, <span class="st">"GSM3039354"</span>, <span class="st">"GSM3039348"</span>, <span class="st">"GSM3039349"</span>, <span class="st">"GSM3039350"</span><span class="op">)</span>, baseDir <span class="op">=</span> <span class="va">data_dir</span>, makeDirectory <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p>Above command should download files to <code>data_dir</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="va">bed_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span>
  path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>,
  pattern <span class="op">=</span> <span class="st">"*bed\\.gz$"</span>,
  full.names <span class="op">=</span> <span class="cn">TRUE</span>, 
  recursive <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">bed_files</span><span class="op">)</span><span class="op">)</span></pre></div>
</div>
</div>
<div id="importing-with-methrix" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-with-methrix" class="anchor"></a>Importing with methrix</h1>
<div id="cpg-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#cpg-annotation" class="anchor"></a>CpG annotation</h2>
<p>As a first step, we need a list of CpG sites in the respective reference genome. The CpG sites are extracted using the respective <a href="https://bioconductor.org/packages/release/bioc/html/BSgenome.html">Bsgenome</a> annotation package. The <code>read_bedgraph</code> function is also able to extract CpG sites on it’s own, however, it might be beneficial to do it separately.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">hg19_cpgs</span> <span class="op">&lt;-</span> <span class="fu">methrix</span><span class="fu">::</span><span class="fu"><a href="../reference/extract_CPGs.html">extract_CPGs</a></span><span class="op">(</span>ref_genome <span class="op">=</span> <span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span><span class="op">)</span></pre></div>
</div>
<div id="sample-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#sample-annotation" class="anchor"></a>Sample annotation</h2>
<p>An annotation table is also necessary to perform analyses. The data will be added to the methrix object, as a <code>colData</code> slot.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">sample_anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span>
    pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"GSM[[:digit:]]*_BiSeq_(WT_D[[:digit:]]{1,2}_R[[:digit:]]).bed.gz"</span><span class="op">)</span>,
    replacement <span class="op">=</span> <span class="st">"\\1"</span>,
    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">bed_files</span><span class="op">)</span>
    <span class="op">)</span>,
  Day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span>
    pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"GSM[[:digit:]]*_BiSeq_WT_(D[[:digit:]]{1,2})_R[[:digit:]].bed.gz"</span><span class="op">)</span>,
    replacement <span class="op">=</span> <span class="st">"\\1"</span>,
    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">bed_files</span><span class="op">)</span>
    <span class="op">)</span>,
  Replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span>
    pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"GSM[[:digit:]]*_BiSeq_WT_D[[:digit:]]{1,2}_(R[[:digit:]]).bed.gz"</span><span class="op">)</span>,
    replacement <span class="op">=</span> <span class="st">"\\1"</span>,
    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">bed_files</span><span class="op">)</span>
    <span class="op">)</span>,
  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">sample_anno</span><span class="op">)</span></pre></div>
</div>
<div id="reading-bedgraph-files" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-bedgraph-files" class="anchor"></a>Reading bedGraph files</h2>
<p><code>read_bedgraphs</code> function is a versatile bedgraph reader intended to import bedgraph files generated virtually by any sort of methylation calling program. It requires user to provide indices for chromosome names, start position and other required fields. However, there are also presets available to import <code>bedgraphs</code> from most common programs such as</p>
<ul>
<li>
<code>Bismark</code> (.cov format)</li>
<li><code>MethylDackel</code></li>
<li><code>MethylcTools</code></li>
<li><code>BisSNP</code></li>
<li><code>BSseeker2_CGmap</code></li>
</ul>
<p>In case if the bedGraphs are generated using any of the above programs use <code>pipiline</code> argument in <code><a href="../reference/read_bedgraphs.html">read_bedgraphs()</a></code> to specify whihc causes the function to automatically infer the requored column indices.</p>
<p>For the example in hand, since we do not know the program used to generate the files, it might worth to take a look at one of the files, and manually specify the column indices.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="va">bed_files</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, nrows <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></pre></div>
<p>The <code>read_bedgraphs</code> function adds CpGs missing from the reference set, and creates a methylation/coverage matrices. Once the process is complete - it returns an object of class <code>methrix</code> which in turn inherits <code>SummarizedExperiment</code> class. <code>methrix</code> object contains ‘methylation’ and ‘coverage’ matrices (either in-memory or as on-disk HDF5 arrays) along with pheno-data and other basic info. This object can be passed to all downstream functions for various analysis. For further details on the data structure, see the <code>SummarizedExperiment</code> package.</p>
<ul>
<li><p>For computers with large memory, increase speed using <code>vect=TRUE</code>. The function will open multiple files, defined by the <code>vect_batch_size</code> argument. Keep in mind, that increasing the batch size will increase the memory need as well.</p></li>
<li><p>If <code>vect=FALSE</code>, the function will iterate along the list of files, keeping only one open at a time.</p></li>
</ul>
<p>A more detailed description of the <code>read_bedgraphs</code> function is here: <a href="https://compepigen.github.io/WGBS_best_practice/read_bedgraphs.html">read_bedgraphs</a>)</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="va">meth</span> <span class="op">&lt;-</span> <span class="fu">methrix</span><span class="fu">::</span><span class="fu"><a href="../reference/read_bedgraphs.html">read_bedgraphs</a></span><span class="op">(</span>
  files <span class="op">=</span> <span class="va">bed_files</span>,
  ref_cpgs <span class="op">=</span> <span class="va">hg19_cpgs</span>,
  chr_idx <span class="op">=</span> <span class="fl">1</span>,
  start_idx <span class="op">=</span> <span class="fl">2</span>,
  M_idx <span class="op">=</span> <span class="fl">5</span>,
  U_idx <span class="op">=</span> <span class="fl">6</span>,
  stranded <span class="op">=</span> <span class="cn">FALSE</span>,
  zero_based <span class="op">=</span> <span class="cn">TRUE</span>, 
  collapse_strands <span class="op">=</span> <span class="cn">FALSE</span>, 
  coldata <span class="op">=</span> <span class="va">sample_anno</span>,
  vect <span class="op">=</span> <span class="cn">TRUE</span>,
  n_threads <span class="op">=</span> <span class="fl">3</span> 
<span class="op">)</span></pre></div>
</div>
<div id="gotchas" class="section level2">
<h2 class="hasAnchor">
<a href="#gotchas" class="anchor"></a>Gotchas</h2>
<p>Too many missing CpGs, if not expected, might indicate that something went wrong.</p>
<ul>
<li><p>An obvious error is using the wrong reference genome.</p></li>
<li><p>A common mistake is falsely indicating strandedness and zero or one based positioning. The combination of the two might lead to only using C-s from one strand (due to a 1 bp shift), resulting lower coverage for the sites. A nice description of the zero- and one-based coordination system can be found <a href="http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/">here</a></p></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Anand Mayakonda, Reka Toth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
